File: Backend/Models/Diagnosis/Patient.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models.Diagnosis
{
    public class Patient
    {
        [Key]
        public int Id { get; set; }

        public User UserProfile { get; set; }

        public List<Condition> Conditions { get; set; }

        public List<Allergy> Allergies { get; set; }

        public List<Surgery> Surgeries { get; set; }

        public List<Medication> CurrentMedications { get; set; }

        public List<Vitals> VitalHistory { get; set; }
        public List<Assessment> Assessments { get; set; }
    }
}

File: Frontend/CloudCure-Angular/src/app/AngularModels/Patient.ts
import { Allergy } from "./Allergy";
import { Assessment } from "./Assessment";
import { Condition } from "./Condition";
import { Medication } from "./Medication";
import { Surgery } from "./Surgery";
import { Vitals } from "./Vitals";
import { UserProfile } from "./UserProfile";
import { EmployeeInformation } from "./EmployeeInformation";
import { Diagnosis } from "./Diagnosis";

export interface Patient 
{
    id?:                 number;
    userProfile:         UserProfile;
    patientName?:         string;
    patientPhone?:        string;
    patientAddress?:      string;
    dateOfBirth?:        Date;
    conditions?:         Condition[];
    allergies?:          Allergy[];
    surgeries?:          Surgery[];
    currentMedications?: Medication[];
    doctor?: EmployeeInformation | undefined;
    diagnoses?: Diagnosis[];
}
